src/csharp/Microsoft.Spark/Interop/Ipc/PayloadHelper.cs

```diff
@@ -183,6 +185,16 @@ internal class PayloadHelper
                                 destination.Position = posAfterEnumerable;
                                 break;
 
+                            case IEnumerable<GenericRow> argRowEnumerable:
+                                SerDe.Write(destination, s_rowTypeId);
+                                SerDe.Write(destination, (int)argRowEnumerable.Count());
```

Comment:
I want the number of rows in argRowEnumerable before I iterate through them all individually as I need to write that number in the destination stream first and then the individual row items. This is because on the scala side I need that number in the stream first so I can create an array of required number of elements. That is not the case in other IEnumerable cases I think, please correct me if I am wrong.

Reply:
The IEnumerable cases seems to do this as well.  Please look at the implementation and verify.

Reply:
Understood, will make the change, thanks!